src/views/SignTransactionLedger.vue

```diff
@@ -201,19 +243,19 @@ export default class SignTransactionLedger extends Vue {
             signedTransaction = await LedgerApi.signTransaction({
                 sender: (senderContract || signer).address,
                 senderType: senderContract ? senderContract.type : Nimiq.Account.Type.BASIC,
-                recipient: this.request.recipient,
-                value: this.request.value,
-                fee: this.request.fee || 0,
+                recipient,
+                value,
+                fee: fee || 0,
                 validityStartHeight,
                 network: Config.network,
-                extraData: this.request.data,
-                flags: this.request.flags,
+                extraData: data,
+                flags,
             }, signer.path, senderAccount.keyId);
         } catch (e) {
             // If cancelled, handle the exception. Otherwise just keep the error message displayed in ledger ui.
             if (e.message.toLowerCase().indexOf('cancelled') !== -1) {
-                if (this.request.kind === RequestType.CHECKOUT) {
-                    this._back(); // user might want to choose another account or address
+                if (this.request.kind === RequestType.CHECKOUT || this.request.kind === RequestType.CASHLINK) {
+                    this._back(); // user might want to choose another account or address or change cashlink
```

Comment:
Does going back to the cashlink form have the values prefilled again? Or do the value and message get lost when going back?

Reply:
They're lost.

Reply:
Understood. So when does this `cancelled` error happen? When the user cancels the action on the Ledger itself? Or by clicking a cancel button in the UI? What happens if the Ledger times out, does it then also do this and go back to the cashlink route?

Reply:
When cancelled on the ledger or when cancelled via code which is triggered, when the view gets destroyed.
I added a change that disables handling the error when the view already is destroyed.
This did not cause any problems so far, but it's cleaner to handle this.

Reply:
Ok. My problem is then that when I cancel it on the Ledger, instead of being able to directly retry, all my Cashlink form input is lost and I have to enter the value and message again. Correct?

I would rather that when I cancel on the Ledger, I get a retry screen and only when I click the arrow back on that screen, I get back to the Cashlink form.

Reply:
But why would a user cancel but then still want to sign the same transaction?

The proper solution would be that the cashlink create form should remember the data when coming back from ledger or keyguard.

Reply:
Yes, you are propably right, if there is no additional user-action between clicking the "create cashlink" button and the start of the Ledger process.

Making the form remember the values would indeed be preferrable, I agree. Should even be possible, because the cashlink is stored in the static store, right?

Reply:
The form does remember the values now.